src/main/java/org/jitsi/jicofo/MeetExtensionsHandler.java

```diff
@@ -87,77 +83,92 @@ public void init()
             = focusManager.getOperationSet(
                     OperationSetDirectSmackXmpp.class).getXmppConnection();
 
-        connection.addAsyncStanzaListener(this, this);
+        muteIqHandler = new MuteIqHandler();
+        colibriIqHandler = new ColibriIqHandler();
+        dialIqHandler = new DialIqHandler();
+        connection.registerIQRequestHandler(muteIqHandler);
+        connection.registerIQRequestHandler(colibriIqHandler);
+        connection.registerIQRequestHandler(dialIqHandler);
     }
 
-    /**
-     * Disposes this instance and stop listening for extensions packets.
-     */
-    public void dispose()
+    private class MuteIqHandler extends AbstractIqRequestHandler
     {
-        if (connection != null)
+        MuteIqHandler()
         {
-            connection.removeAsyncStanzaListener(this);
-            connection = null;
+            super(MuteIq.ELEMENT_NAME, MuteIq.NAMESPACE, IQ.Type.get, Mode.sync);
         }
-    }
 
-    @Override
-    public boolean accept(Stanza packet)
-    {
-        return acceptMuteIq(packet)
-                || acceptColibriIQ(packet)
-                || acceptRayoIq(packet)
-                || acceptPresence(packet);
+        @Override
+        public IQ handleIQRequest(IQ iqRequest)
+        {
+            return handleMuteIq((MuteIq) iqRequest);
+        }
     }
 
-    @Override
-    public void processStanza(Stanza packet)
+    private class ColibriIqHandler extends AbstractIqRequestHandler
     {
-        if (connection == null)
+        ColibriIqHandler()
         {
-            logger.error("Not initialized");
-            return;
+            super(ColibriConferenceIQ.ELEMENT_NAME,
+                ColibriConferenceIQ.NAMESPACE,
+                IQ.Type.get,
+                Mode.sync);
```

Comment:
I don't think we want to use Mode.sync here. If you take a look at "QueuePacketProcessor" which was used here before it was putting on a separate queue to not block the main packet reader thread. I'm guessing that the "async" mode will do the same here.

Reply:
I don't mind changing this to async, but I wonder why this would be necessary. This handler is basically doing nothing (check if the get IQ has a recording set, check if there's a matching conference, return the result IQ).